src/t_zset.c

```diff
@@ -3819,6 +3822,12 @@ void genericZpopCommand(client *c, robj **keyv, int keyc, int where, int emitkey
         return;
     }
 
+    if (reply_empty_with_zero_count && count == 0) {
+        /* ZPOPMIN/ZPOPMAX with count 0. */
+        addReply(c, shared.emptyarray);
+        return;
+    }
+
```

Comment:
i don't like that argument (`reply_empty_with_zero_count`).
it suggests that when it is false, the reply will not be an empty array (e.g. a NIL or something).
we already have other arguments that that define the type of reply (`use_nested_array`, `reply_nil_when_empty`)

but in fact IIUC, this new arg just says that there are two types of `count==0` (it's not about the reply, but rather just define the input better)
* one means single item (i.e. a special mode of count of `1` that doesn't return an array)
* the other is a request for multiple items (array) with count of 0.

the simple solution here is to use a different value for "single" count.
i.e. use `-1` instead of `0`, and then we don't need any new arguments.